js-packages/@prestojs/rest/src/CursorPaginator.ts

```diff
@@ -0,0 +1,116 @@
+import { EndpointExecuteOptions } from './Endpoint';
+import Paginator, { PaginationStateChangeListener } from './Paginator';
+
+export type CursorPaginationState = {
+    pageSize?: number;
+    cursor?: string;
+};
+
+export default class CursorPaginator extends Paginator {
+    nextCursor: string | null = null;
+    previousCursor: string | null = null;
+
+    get cursor(): string | null | undefined {
+        return this.currentState.cursor;
+    }
+
+    get pageSize(): number | null | undefined {
+        return this.currentState.pageSize;
+    }
+
+    constructor(
+        initialState: CursorPaginationState = {},
+        onChange?: PaginationStateChangeListener
+    ) {
+        super(initialState || {}, onChange);
+    }
+
+    setPageSize(pageSize: number): void {
+        if (pageSize < 1) {
+            throw new Error(`Invalid pageSize ${pageSize} - should be >= 1`);
+        }
+        this.setState(currentState => {
+            // If page size changes any next/previous cursors are no invalid
```

Comment:
I'm pretty sure I just meant in terms of consistent page sizes it's no longer valid (ie. it now points to something part way down a page, so if you go previous back to the start you'll get an overlap on page 1 and page 2)

Reply:
I think it should clear next and previous as they are relative to cursor at old page size... it shouldn't matter because if you change page size it should refetch and update them anyway

Reply:
I've left the code that clears next/previous on change but it will now retain the current cursor on page size change